libparsec/crates/client/src/workspace/store/mod.rs

```diff
@@ -562,4 +573,65 @@ impl WorkspaceStore {
             .await
             .map_err(UpdateRealmCheckpointError::Internal)
     }
+
+    pub async fn apply_prevent_sync_pattern(&self) -> Result<(), ApplyPreventSyncPatternError> {
+        let mut storage_guard = self.storage.lock().await;
+        let storage = storage_guard
+            .as_mut()
+            .ok_or(ApplyPreventSyncPatternError::Stopped)?;
+
+        let (pattern, applied) = storage
+            .get_prevent_sync_pattern()
+            .await
+            .map_err(ApplyPreventSyncPatternError::Internal)?;
+        if applied {
+            return Ok(()); // Already applied
+        }
+
+        recursive_apply_prevent_sync_pattern(storage, self.realm_id, &pattern, &self.device)
+            .await?;
+        storage
+            .mark_prevent_sync_pattern_fully_applied(&pattern)
+            .map_err(ApplyPreventSyncPatternError::Internal)
+            .await?;
+        Ok(())
+    }
+}
+
+fn recursive_apply_prevent_sync_pattern(
+    storage: &mut WorkspaceStorage,
+    entry_id: VlobID,
+    pattern: &Regex,
+    device: &LocalDevice,
+) -> Box<dyn Future<Output = Result<(), ApplyPreventSyncPatternError>>> {
```

Comment:
TODO Handle RAM cache + DB cache.

I should use `CurrentViewCache`.

- Should I first try to update a manifest in the cache then update the database ?
- Only update the database and indicate the cache to load the data from the DB ?